tests/lib/nsxv_test.py

```diff
@@ -747,108 +794,156 @@ def testBadHeaderCase4(self):
+                                                         MIXED_TO_MIXED)
+    self.assertFalse(len(source_addr) == 0)
+    for saddr in source_addr:
+      saddr_type = saddr.find('type').text
+      self.assertIn(saddr_type, ('Ipv4Address', 'Ipv6Address'))
 
-    self.naming.GetNetAddr.assert_has_calls(
-        [mock.call('GOOGLE_DNS'), mock.call('SOME_HOST')])
+    self.assertFalse(len(dest_addr) == 0)
+    for daddr in dest_addr:
+      daddr_type = daddr.find('type').text
+      self.assertIn(daddr_type, ('Ipv4Address', 'Ipv6Address'))
 
-  def testNsxvStrForDestinationIpv4Mismatch(self):
+    self.naming.GetNetAddr.assert_has_calls([mock.call('GOOGLE_DNS')] * 2)
+
+  def testMixedToAny(self):
     self.naming.GetNetAddr.side_effect = [
-        [nacaddr.IP('8.8.4.4'),
-         nacaddr.IP('8.8.8.8'),
+        [nacaddr.IP('8.8.4.4'), nacaddr.IP('8.8.8.8'),
          nacaddr.IP('2001:4860:4860::8844'),
-         nacaddr.IP('2001:4860:4860::8888')],
+         nacaddr.IP('2001:4860:4860::8888')]]
+
+    source_addr, dest_addr = self.getSourceDestAddresses(MIXED_HEADER +
+                                                         MIXED_TO_ANY)
+    self.assertFalse(len(source_addr) == 0)
+    for saddr in source_addr:
+      saddr_type = saddr.find('type').text
+      self.assertIn(saddr_type, ('Ipv4Address', 'Ipv6Address'))
+
+    self.assertTrue(len(dest_addr) == 0)
+    self.naming.GetNetAddr.assert_has_calls([mock.call('GOOGLE_DNS')])
+
+  def testV4ToV4(self):
+    self.naming.GetNetAddr.side_effect = [
+        [nacaddr.IP('10.0.0.1'), nacaddr.IP('10.0.0.2')],
+        [nacaddr.IP('10.0.0.0/8'), nacaddr.IP('172.16.0.0/12'),
+         nacaddr.IP('192.168.0.0/16')]]
+
+    source_addr, dest_addr = self.getSourceDestAddresses(MIXED_HEADER +
+                                                         V4_TO_V4)
+    self.verifyAddressType(source_addr, 'Ipv4Address')
+    self.verifyAddressType(dest_addr, 'Ipv4Address')
+    self.naming.GetNetAddr.assert_has_calls(
+        [mock.call('NTP_SERVERS'), mock.call('INTERNAL')])
+
+  def testV6ToV6(self):
+    self.naming.GetNetAddr.side_effect = [
+        [nacaddr.IP('2001:4860:8000::/33')],
+        [nacaddr.IP('2001:4860:8000::/33')]]
+
+    source_addr, dest_addr = self.getSourceDestAddresses(MIXED_HEADER +
+                                                         V6_TO_V6)
+    self.verifyAddressType(source_addr, 'Ipv6Address')
+    self.verifyAddressType(dest_addr, 'Ipv6Address')
+    self.naming.GetNetAddr.assert_has_calls(
+        [mock.call('SOME_HOST')] * 2)
+
+  def testV4ToV6(self):
+    self.naming.GetNetAddr.side_effect = [
+        [nacaddr.IP('10.0.0.0/8'), nacaddr.IP('172.16.0.0/12'),
+         nacaddr.IP('192.168.0.0/16')],
         [nacaddr.IP('2001:4860:8000::/33')]]
-    pol = policy.ParsePolicy(MIXED_HEADER + DEST_IPv4_MISMATCH, self.naming,
-                             False)
+
+    source_addr, dest_addr = self.getSourceDestAddresses(MIXED_HEADER +
```

Comment:
This shouldn't return a term because it drops both sides (and it correctly does). This test is relying on the return is 0 based on the search of the xml. Perhaps add a comment to make it explicit that the expectation is that no term gets generated. 

In fact I think it might be useful to have an explicit check that no term was created, however it's up to you I won't require it.